sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/Permission.java

```diff
@@ -112,9 +113,10 @@ public PartitionKey getResourcePartitionKey() {
      * @param partitionkey the partition key.
      */
     public void setResourcePartitionKey(PartitionKey partitionkey) {
-        BridgeInternal.setProperty(this,
-                                   Constants.Properties.RESOURCE_PARTITION_KEY,
-                                   BridgeInternal.getPartitionKeyInternal(partitionkey)
-                                       .toJson());
+        if(ModelBridgeInternal.getPartitionKeyObject(partitionkey) != null) {
```

Comment:
why the null check is needed? 

can the user pass null partition key to the setter? is that a valid scenario?
if we decide to do arg validation here shouldn't we throw NPE?

Reply:
I agree, good point, having null checks prevent bugs, but they also hide the possible buggy code paths. We should avoid it if this scenario can be validated through the generic code pass.